### DISCUSSION TITLE
pfSense/OPNsence VM (and other iso files)

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1937] [Announcements]
I find the process to be quite straightforward and don't believe it requires a script.

1. Begin by obtaining pfSense. In the Proxmox shell, execute the following command:

(US Mirror)
```
wget -q --show-progress -O - https://atxfiles.netgate.com/mirror/downloads/pfSense-CE-2.7.0-RELEASE-amd64.iso.gz | gzip -d -c > /var/lib/vz/template/iso/pfSense-CE-2.7.0-RELEASE-amd64.iso
```
(Germany Mirror)
```
wget -q --show-progress -O - https://frafiles.netgate.com/mirror/downloads/pfSense-CE-2.7.0-RELEASE-amd64.iso.gz | gzip -d -c > /var/lib/vz/template/iso/pfSense-CE-2.7.0-RELEASE-amd64.iso
```
2. Create a virtual machine manually and choose the downloaded ISO file as the installation source.
3. Complete the setup by accessing the virtual machine's console.

**Comment by endigma** [comment]
What about OPNsense?

**Reply by tteck** [reply]
```
wget -q --show-progress -O - https://mirror.wdc1.us.leaseweb.net/opnsense/releases/mirror/OPNsense-23.7-dvd-amd64.iso.bz2 | bzip2 -d -c > /var/lib/vz/template/iso/OPNsense-23.7-dvd-amd64.iso
```

**Reply by endigma** [reply]
I wrote this thinking it was a script post, should have read all the way I guess

**Comment by Dvalin21** [comment]
Im a little confused. Is this just installing it as a VM?  Is it possible to install as a lxc?  I know between choosing VM and lxc, its 1000 times easier passing pcie devices to an lxc than it is for vm. And for some computers/servers its impossible to the vm because depending on your hardware it may not support separate groupings when

**Reply by endigma** [reply]
No, an LXC is a linux container, pf/opn are BSD

**Reply by Dvalin21** [reply]
Thanks for clearing that up. It would be nice if they could do the same for BSD
